### PR TITLE
Limit mouse move events to one every 17 mS.

### PR DESCRIPTION
**Pull request by uklatt**
Limit mouse move events to one every 17 mS.

### CODE REVIEW

File: core/rfb.js [hunk 1]
```diff
@@ -790,8 +797,15 @@ export default class RFB extends EventTargetMixin {
 
         if (this._viewOnly) { return; } // View only, skip mouse events
 
-        if (this._rfb_connection_state !== 'connected') { return; }
-        RFB.messages.pointerEvent(this._sock, this._display.absX(x), this._display.absY(y), this._mouse_buttonMask);
+        // Limit mouse move events to one every MOUSE_MOVE_DELAY mS
+        clearTimeout(this.timerID);     // Remove old one
```
Comment (CendioOssman) [inline core/rfb.js]: `timerID` is a bit generic. Please include something that identifies what timer it is.

File: core/rfb.js [hunk 2]
```diff
@@ -790,8 +797,15 @@ export default class RFB extends EventTargetMixin {
...
+        this.newMouseMoveTime = Date.now();
```
Comment (CendioOssman) [inline core/rfb.js]: This can be local as far as I can tell.

File: core/rfb.js [hunk 3]
```diff
@@ -790,8 +797,15 @@ export default class RFB extends EventTargetMixin {
...
+        if ( this.newMouseMoveTime < this.oldMouseMoveTime + MOUSE_MOVE_DELAY ) {
+            this.timerID = setTimeout(this._setMousePosition, MOUSE_MOVE_DELAY, x, y);  // Set last mouse position after timeout
```
Comment (CendioOssman) [inline core/rfb.js]: I don't think this works properly. `this` would be incorrectly set when this timeout fires. Have you tested it?

On that note, some unit tests would be nice for this.

**Comment by samhed** [comment]
Ping!

This would be nice to get merged, is there anything we can do to help get this moving forward?

File: core/rfb.js (hunk 1 above)
Comment (samhed) [reply core/rfb.js]: fixed

**Comment by samhed** [comment]
I moved the code to `core/input/mouse.js`, combined the commits, and fixed the comments by @CendioOssman.

File: core/rfb.js (hunk 3 above)
Comment (samhed) [reply core/rfb.js]: fixed bind()

and added unit tests in a672168d4dac5735b40cfe618a47f76b69c85882

**Pull request merged**